### PR TITLE
Faster post build startup:

### PR DESCRIPTION
**Pull request by KevinMind**
Fixes: mozilla/addons#15239

### Description

- use named volumes that mount faster and don't need renewing
- skip docker compose healthcheck
- add healthcheck script to initialize

### Context

Faster post build by parallelizing initialize and startup

### Verify time of startup

You can run the steps of the build independently to get a feel for relative times.

1. Build: setup and building the image

Try building the image first without cache.

> [!WARNING]
> This is an aggressive command that will clear local images/volumes and docker state

```bash
make clean_docker
```

> Make sure you don't have any dangling volumes or images.. cleaning up docker can be a chore....

```bash
time make up_pre
```

Check the time, then run the command again. The second time should be much, much faster.

Try it after changing a file in `./src` this will invalidate some of the cache and give a more realistic cached time.

2. Start: starting the containers

Now start the containers, running

```bash
time make up_start
```

The first time, it should take some time to download images. Run it again and it should be quite fast.

> [!NOTE]
> Before running the second time, run `make down` so that the containers are not running but are pulled.

3. Initialize: setting up container state and waiting to be healthy

Now run the initialize command to initialize the container state and ensure everything is healthy.

```bash
time make up_post
```

If you have a truly clean environment this could take some time as it must migrate and populate the database.  Run it again and it should be relatively fast.

### Verify no anonymous or unnamed volumes

You can independently verify the non existence of anonymous volumes via this query that returns volumes from all containers that are:
- bind mount (anonymous volumes)
- volume mount missing source (no name)

```bash
docker compose config | yq -r '.services.*.volumes[]? | select((.type == "bind") or (has("source") | not))'
```

### Checklist

- [X] Add `#ISSUENUM` at the top of your PR to an existing open issue in the mozilla/addons repository.
- [X] Successfully verified the change locally.
- [x] The change is covered by automated tests, or otherwise indicated why doing so is unnecessary/impossible.
- [ ] Add before and after screenshots (Only for changes that impact the UI).
- [ ] Add or update relevant [docs](../docs/) reflecting the changes made.

### CODE REVIEW

**Comment by KevinMind** [comment]
1. make up_pre  5.21s user 5.43s system 4% cpu 3:58.51 total
2. make up_pre  0.63s user 0.50s system 44% cpu 2.541 total
3. make up_start  0.71s user 0.52s system 0% cpu 2:15.34 total

**Comment by KevinMind** [comment]
> I got a little lost during the test steps a few times so I don't really have comparable figures for all the steps, but my `time make up` after a `make down` and changing a file in `./src/ is
> 
> real 1m52.840s user 0m2.189s sys 0m1.734s
> 
> which is somewhat of an improvement - but not quite as speedy as your figures.
> 
> fwiw, `[+] Building 45.6s (29/29) FINISHED` - so the docker part is 45 seconds; the rest is the django initialization (even with no migrations, it takes a little bit of time) + es reindexing, pretty much.

Yeah the goal of this PR is to make the 2 and 3 of the 3 steps faster. It doesn't really impact the build itself. The "better" way of measuring the impact here would be to make the changes to the make recipes first, then time the original state, and then add the changes on this patch so you can get per phase timing... It's faster. At the end of this epic I plan to test the July state of the code to the end state and we can get some real data on the impact.

File: .github/workflows/_test_check.yml [hunk 1]
```diff
@@ -65,7 +65,8 @@ jobs:
             version: ${{ matrix.version }}
             compose_file: ${{ matrix.compose_file }}
           EOF
-      - uses: ./.github/actions/run-docker
+      - name: ${{ matrix.version == 'local' && 'Uncached Build' || 'Pull' }} Check
```
[the remark this answers is not in the document]
Comment (KevinMind) [reply .github/workflows/_test_check.yml:68]: This is pretty standard JS syntax afaik ([GHA operators](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#operators)), it's the first one. The statements are evaluated left to right.

image

[the remark this answers is not in the document]
Comment (eviljeff) [reply .github/workflows/_test_check.yml:68]: Ah, it's `'Uncached build' if matrix.version == 'local' else 'Pull'` in python syntax

**Pull request merged**